tokio/src/task/local.rs

```diff
@@ -512,6 +491,21 @@ impl LocalSet {
         run_until.await
     }
 
+    pub(in crate::task) fn spawn_named<F>(
+        &self,
+        future: F,
+        name: Option<&str>,
+    ) -> JoinHandle<F::Output>
+    where
+        F: Future + 'static,
+        F::Output: 'static,
+    {
+        let handle = self.context.spawn(future, name);
+
+        self.context.shared.waker.wake();
```

Comment:
Perhaps it would be better to put the call to `wake` inside `context.spawn`? It's quite important that we don't forget to call it in one of the spawn calls.

Reply:
The free-function version of `spawn_local` does _not_ call `wake()` previously; only spawning through a `LocalSet` reference calls `wake()`:

https://github.com/tokio-rs/tokio/blob/ddf2f5fdf67691cf4b057343ad4dd0759730d12d/tokio/src/task/local.rs#L299-L310

I believe this is because the free function can only spawn tasks while already executing on the `LocalSet`, so the `LocalSet` has already been woken. I didn't put the wake call in `Context::spawn` because I wanted to preserve the current behavior.

Reply:
Aha... Well, that certainly needs some better comments.

Reply:
i can add one now.